web/pdf_page_view.js

```diff
@@ -452,11 +452,41 @@ class PDFPageView {
 
     let resultPromise = paintTask.promise.then(function() {
       return finishPaintTask(null).then(function () {
+        let pump;
```

Comment:
That just suggest, at least to me, that further changes may be required (perhaps in `text_layer_builder.js`).
However, if it turns out to be absolutely necessary to fetch *all* textContent here anyway, I really don't understand why all this code is needed here. In that case you could just keep the existing `pdfPage.getTextContent` call here (in addition to the `streamTextContent` one), to aid readability of this code!